src/IPCQueue.h

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include <FreeRTOS.h>
+#include <queue.h>
+
+#include <stdint.h>
+
+/*
+ * IPCMessage
+ */
+
+enum IPCMessageFlags {
+  IPC_MESSAGE_FLAG_WAIT = 0x01000000,
+  IPC_MESSAGE_FLAG_COPY = 0x02000000
+};
+
+struct IPCMessage
+{
+  /* Two dword for data transfers. */
+  uint32_t mDWord0;
+  uint32_t mDWord1;
+  /* Message flags and buffer length */
+  uint32_t mFlags;
+  /* Message buffer */
+  const void* mBuffer;
+};
+
+int
+IPCMessageInit(struct IPCMessage* aMsg);
+
+int
+IPCMessageProduce(struct IPCMessage* aMsg);
+
+int
+IPCMessageConsume(struct IPCMessage* aMsg);
+
+/*
+ * IPCMessageQueue
+ */
+
+struct IPCMessageQueue
```

Comment:
I'll use a typedef. I'm actually in the pro-struct camp, because typedef obscures important information about the type. But it's not that important.